src/app/src/components/movie/movie.component.spec.ts

```diff
@@ -35,10 +40,9 @@ describe('MovieComponent', () => {
     expect(component.selectedAsFavourite).toBe(false);
   });
 
-  it('should redirect when clicked'),
-    () => {
-      component.showMovieDetails(0);
-
-      expect(component).toBeFalsy();
-    };
+  it('should redirect when clicked', () => {
+    spyOn(component, 'showMovieDetails');
+    component.showMovieDetails(0);
+    expect(component.showMovieDetails).toHaveBeenCalled();
+  });
```

Comment:
What you have done: 
Spy on a method
Call that method
Check if what method was called

What should happen `it('should naviage to movie details page')`
`spyOn(router, 'navigate);`
call method
check `expect(router.navigate).toHaveBeenCalledWith(...);`

Reply:
also above tests for `selectedAsFavourite` can be done in 1 test case like this

```
describe('markMovieAsFavourite', () => {
  it('should set selectedAsFavourite to opposite value', () => {
    component.markMovieAsFavourite();
    expect(component.selectedAsFavourite).toBeTrue();
    component.markMovieAsFavourite();
    expect(component.selectedAsFavourite).toBeFalse();
  })
}) 
```